app/models/user/reputation_acquisition.rb

```diff
@@ -2,7 +2,23 @@ class User::ReputationAcquisition < ApplicationRecord
   belongs_to :user
   belongs_to :reason_object, polymorphic: true, optional: true
 
+  before_create do
+    self.amount = REASON_AMOUNTS[self.reason]
+  end
+
+  validates :reason, inclusion: {
+    in: %i[exercise_authorship exercise_contributorship],
+    message: "%<value>s is not a valid reason",
+    strict: ReputationAcquisitionReasonInvalid
+  }
```

Comment:
Nice! 

Put this before the `before_create`.

I try and keep the order as associations, validations, scopes, callbacks, methods